### PR TITLE
Allowing stop processing at a certain event number.

### PR DESCRIPTION
**Pull request by james-lappin-cko**

### CODE REVIEW

File: src/Ditto/ReplicatingConsumer.cs [hunk 1]
```diff
@@ -43,6 +43,14 @@ public async Task ConsumeAsync(string eventType, ResolvedEvent resolvedEvent)
         {
             if (string.IsNullOrWhiteSpace(eventType)) throw new ArgumentException("Event type required", nameof(eventType));
 
+            if (_settings.EventNumberToStopAt.HasValue && resolvedEvent.OriginalEventNumber > _settings.EventNumberToStopAt.Value)
+            {
+                _logger.Error("Please stop ditto as it has reached the event number to stop at. Skipping processing." +
```
Comment (mikhail-kuzmiankou-cko) [inline src/Ditto/ReplicatingConsumer.cs]: You might want to add a space at the end 🙂

**Review by mikhail-kuzmiankou-cko — APPROVED** [review]

File: src/Ditto/ReplicatingConsumer.cs (hunk 1 above)
Comment (james-lappin-cko) [reply src/Ditto/ReplicatingConsumer.cs]: Thanks

**Comment by mikhail-kuzmiankou-cko** [comment]
Take into account, that the position of persistent subscription will proceed so all consecutive events after the configured position will be marked as handled. You won't be able to re-use this persistence subscription to handle skipped events. But anyway, you can create a new subscription with the desired position to start from.

**Comment by james-lappin-cko** [comment]
Yeah exactly. Which is why I have printed out the logs as much info as I can.

I may need to use the task to unpark events but shouldn't take too long

**Pull request merged**